dms/models/dms_security_mixin.py

```diff
@@ -133,7 +133,12 @@ def _get_domain_by_inheritance(self, operation):
             domains.append([("res_model", "=", model._name), ("res_id", "=", False)])
             # Check record access in batch too
             group_ids = [i for i in group["res_id"] if i]  # Hack to remove None res_id
-            related_ok = model.browse(group_ids)._filter_access_rules_python(operation)
+            # Do a search instead of browse() in case a record does not exist (e.g. a
```

Comment:
If the problem is not existing groups, then you can do `model.browse(group_ids).exists()`, but are you sure this is the problem? The problem was on attachments with an unexisting `res_model` or `res_id` target.

Reply:
The problem is that in some cases the linked directories or files may not have been deleted (e.g. by deleting with SQL in migration scripts), something that is covered in https://github.com/OCA/dms/blob/15.0/dms/models/base.py#L11.

Reply:
But I insist, this one is not preventing that case.

Reply:
This change prevents an error from being displayed when trying to access a record that does not exist (the initial use case), and the test verifies it. If you have any other suggestion, it is welcome.

Reply:
OK, part of my non-understanding is the bad variable name `group_ids`. Please call it `referred_ids` or similar. After that, my initial suggestion `model.browse(referred_ids).exists()` is more compact and may preserve caches.

Reply:
Changes done.